File: packages/authentication/ios/Plugin/Classes/Results/FetchSignInMethodsForEmailResult.swift
import Foundation
import Capacitor

@objc public class FetchSignInMethodsForEmailResult: NSObject, Result {
    let signInMethods: [String]

    init(_ signInMethods: [String]) {
        self.signInMethods = signInMethods
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["providerIds"] = signInMethods
        return result as AnyObject
    }
}

File: packages/authentication/android/src/main/java/app/capgo/capacitor/firebase/authentication/FetchSignInMethodsForEmailResult.java
package app.capgo.capacitor.firebase.authentication.classes.results;

import app.capgo.capacitor.firebase.authentication.interfaces.Result;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import java.util.List;

public class FetchSignInMethodsForEmailResult implements Result {

    private List<String> signInMethods;

    public FetchSignInMethodsForEmailResult(List<String> signInMethods) {
        this.signInMethods = signInMethods;
    }

    public JSObject toJSObject() {
        JSArray signInMethodsResult = new JSArray();
        for (String signInMethod : signInMethods) {
            signInMethodsResult.put(signInMethod);
        }

        JSObject result = new JSObject();
        result.put("signInMethods", signInMethodsResult);
        return result;
    }
}
